lib/pbench/server/api/resources/api_key.py

```diff
@@ -61,17 +118,48 @@ def _post(
             new_key = APIKey.generate_api_key(user)
         except Exception as e:
             raise APIInternalError(str(e)) from e
-
         try:
-            key = APIKey(api_key=new_key, user=user)
+            key = APIKey(api_key=new_key, user=user, name=name)
             key.add()
             status = HTTPStatus.CREATED
         except DuplicateApiKey:
             status = HTTPStatus.OK
         except Exception as e:
             raise APIInternalError(str(e)) from e
-
-        context["auditing"]["attributes"] = {"key": new_key}
-        response = jsonify({"api_key": new_key})
+        context["auditing"]["attributes"] = key.as_json()
+        response = jsonify(key.as_json())
         response.status_code = status
         return response
+
+    def _delete(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """Delete the requested key.
+
+        DELETE /api/v1/key/{key}
+
+        Returns:
+            Success: 200
+
+        Raises:
+            APIAbort, reporting "UNAUTHORIZED" and "NOT_FOUND"
+            APIInternalError, reporting the failure message
+        """
+        key_id = params.uri["key"]
+        user = Auth.token_auth.current_user()
+
+        if not user:
+            raise APIAbort(
+                HTTPStatus.UNAUTHORIZED,
+                "User provided access_token is invalid or expired",
+            )
+        key = APIKey.query(id=key_id, user=user)
+        if not key:
+            raise APIAbort(HTTPStatus.NOT_FOUND, "Requested key not found")
+        key = key[0]
+        try:
+            key.delete()
+            context["auditing"]["attributes"] = {"key_id": key.id, "name": key.name}
```

Comment:
I'd prefer that you use `.as_json()` here, also, so the format is consistent. You're using a different name for the `id` key in this case which is less clear and would also make scripted analysis of the audit log more difficult! (Doing that *before* the `key.delete()` might be cleaner: I'm not sure if SQLAlchemy will be upset with accessing the proxy after that.)

Reply:
> Doing that _before_ the `key.delete()` might be cleaner

As the code stands presently, if the `delete()` _fails_, what do we get in the audit log?

It seems like it might be _kind of important_ to do the assignment before the `delete()` call....  🥴

Reply:
Yeah; moving the `.as_json()` first is more clear and safer.